### PR TITLE
update all ecephys interface channel properties

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
Adaptation to [PR #181](https://github.com/catalystneuro/nwb-conversion-tools/pull/181). I wrote this stuff a while back in anticipation of the changes, but a lot has changed since then so I'll just open it as a draft and update once #181 is through.

## Motivation

Sakhsham's new `add_electrode` method clarifies the boundary of heavy vs. light metadata for electrode tables. All data values that match to electrode columns are now set as channel properties in the `RecordingExtractor` object, while text descriptions are still passed along the usual global `metadata` object.

## How to test the behavior?
```
# TODO
```

## Checklist

- [X] Have you checked our [Contributing](https://github.com/catalystneuro/nwb-conversion-tools/blob/master/docs/contribute.rst) document?
- [X] Have you ensured the PR description clearly describes problem and the solution?
- [X] Is your contribution compliant with our coding style? Read about [PEP8](https://www.python.org/dev/peps/pep-0008/) and [black](https://black.readthedocs.io/en/stable/the_black_code_style.html).
- [X] Have you checked to ensure that there aren't other open or previously closed [Pull Requests](https://github.com/catalystneuro/nwb-conversion-tools/pulls) for the same change? (shares some similarity with #181 for certain interfaces)
- [X] Have you included the relevant issue number using `#XXX` notation where `XXX` is the issue number?

### CODE REVIEW

**Comment by CodyCBakerPhD** [comment]
@luiztauffer @sbuergers Take a look if you get a second - host of updates to various interfaces y'all use with respect to the new electrodes method, feel free to modify as needed

File: nwb_conversion_tools/datainterfaces/ecephys/neuroscope/neuroscopedatainterface.py [hunk 1]
```diff
@@ -80,21 +98,19 @@ def get_ecephys_metadata(xml_file_path: str):
...
-                        description="0-indexed channel within a shank.",
-                        data=[x for channels in shank_channels for x, _ in enumerate(channels)]
+                        description="0-indexed channel within a shank."
                     ),
                     dict(
                         name='group_name',
-                        description="The name of the ElectrodeGroup this electrode is a part of.",
-                        data=[f"shank{n + 1}" for n, channels in enumerate(shank_channels) for _ in channels]
+                        description="The name of the ElectrodeGroup this electrode is a part of."
                     )
```
Comment (Saksham20) [inline nwb_conversion_tools/datainterfaces/ecephys/neuroscope/neuroscopedatainterface.py:114]:
```suggestion
```

File: nwb_conversion_tools/datainterfaces/ecephys/openephys/openephysdatainterface.py [hunk 2]
```diff
@@ -46,10 +46,16 @@ def get_metadata(self):
         session_start_time = self.recording_extractor._fileobj.experiments[0].datetime
         session_start_time_tzaware = pytz.timezone('EST').localize(session_start_time)
 
+        for channel_id in self.recording_extractor.get_channel_ids():
+            self.recording_extractor.set_channel_property(
+                channel_id=channel_id,
+                property_name="group_name",
+                value="ElectrodeGroup"
+            )
+
```
Comment (Saksham20) [inline nwb_conversion_tools/datainterfaces/ecephys/openephys/openephysdatainterface.py:55]:
```suggestion
```

File: nwb_conversion_tools/datainterfaces/ecephys/spikeglx/spikeglxdatainterface.py [hunk 3]
```diff
@@ -27,19 +27,54 @@ def fetch_spikeglx_metadata(file_path: str, recording: RecordingExtractor, metad
...
+        ],
+        Electrodes=[
+            dict(
+                name='shank_electrode_number',
+                description="0-indexed channel within a shank."
+            ),
+            dict(
+                name='group_name',
+                description="The name of the ElectrodeGroup this electrode is a part of."
+            )
```
Comment (Saksham20) [inline nwb_conversion_tools/datainterfaces/ecephys/spikeglx/spikeglxdatainterface.py:75]:
```suggestion
```

**Review by Saksham20 — CHANGES_REQUESTED** [review]

**Comment by CodyCBakerPhD** [comment]
Attempting to update master into this results in as many conflicts as there are changes - would just be easier to reopen under different branch. I'll try to get around to this soon

**Pull request closed without merging**